### PR TITLE
Refactoring of compute_performance.

### PR DESCRIPTION
**Pull request by samuelgarcia**
See #77.

Refactore totally the compute performance to allow sevral counting strategies.
This should avoid bias with some strategy and open the workflow to other couting methods.

@colehurwitz31 @magland @mhhennig @alejoe91 : could you have a look ?

### CODE REVIEW

**Comment by colehurwitz31** [comment]
I can't comment too much on your refactoring since I don't really understand the agreement code, however, I fixed all of the bugs unrelated to your pull request.

Now there is just a key error in the sorting comparison I believe.

**Comment by samuelgarcia** [comment]
@colehurwitz31 : I have just fix problems.
Travis is now OK.
Thank for fixes in hearding spike.

I have update a full notebook to demonstrate how we can compare **sevral** sorter on **several** with **several** metric method.
I am really happy with this new workflow. I wanted to do this since mounth.
I hope I don't break to much old code.

**Comment by samuelgarcia** [comment]
@alejoe91: 
If you review this code, here a summary to anderstand the changes:
  * **do_counting()** have been splitted in 2 parts: **do_score_labels()** + **do_counting()**
  * **do_counting()** make several counting
  * **compute_performance())** is now a method **SortingCOmparison.get_performance**
  * **get_performance())** have **method='by_spiketrain' or 'pooled_with_sum' or 'pooled_with_average'**
  * **gather_sorting_comparison** have been enhanced.

I have a doubt about the naming of methods thenself and name of dataframes in gather_sorting_comparison() many theses names are too long. Tell me.

**Comment by colehurwitz31** [comment]
Will this be combined with Matthias's parallel do_matching pull request? Are they compatible?

**Comment by samuelgarcia** [comment]
@colehurwitz31 : normally yes. If there are some conflict, I will be able to resolve manually.

**Comment by alejoe91** [comment]
Ok this looks great. I'll review both this and Matthias tomorrow afternoon. Then we can merge one and solve any conflict arising by merging the other. :)

**Comment by samuelgarcia** [comment]
Since this PR also make travis happy I propose to merge it fisrt, no ?

**Comment by samuelgarcia** [comment]
@alejoe91 : note that in this PR I also modify a lot the notebook example on comparison adding all new feature available here. this make #98 not compatible with this for noetbook update.

**Comment by alejoe91** [comment]
Yep sounds good! I think that in the comparison notebook we can put both simple pariwise comparisons, multiple sorting, and performance comparison with ground truth. I haven't made any changes there apart from the title

File: spiketoolkit/comparison/sortingcomparison.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import pandas as pd
 
 from .comparisontools import (count_matching_events, compute_agreement_score,
-                                                do_matching, do_counting, do_confusion_matrix)
+                                                do_matching, do_score_labels, do_counting, do_confusion_matrix)
```
Comment (alejoe91) [inline spiketoolkit/comparison/sortingcomparison.py]: @samuelgarcia could you make a function `compare_sorters` that returns a `SortingComparison` object?
I think that it is in line with the preprocessing module and easier to use. Then we only have to add docstring for the function.

**Comment by samuelgarcia** [comment]
THank you!

I will add the function.

**Comment by alejoe91** [comment]
No worries! I'm adding it to Matthias' PR together with multisorting.

`compare_two_sorters`
`compare_multiple_sorters`

**Pull request merged**